engine/src/cuDF/generator/sample_generator.cu

```diff
@@ -72,10 +72,10 @@ std::unique_ptr<ral::frame::BlazingTable> generate_sample(
 
 	cudf::generator::RandomVectorGenerator<std::int32_t> generator(0L, num_rows);
 	std::vector<std::int32_t> arrayIdx = generator(num_samples);
-
+	
 	rmm::device_buffer gatherData(arrayIdx.data(), num_samples * sizeof(std::int32_t));
 
-	cudf::column gatherMap{cudf::data_type{cudf::type_id::INT32}, num_samples, gatherData};
+	cudf::column gatherMap{cudf::data_type{cudf::type_id::INT32}, num_samples, std::move(gatherData)};
```

Comment:
what will happen to this code when the indices change from and int32 to an int64. Is there a way of doing this where we tie it to whatever cudf::sizetype is? like whatever we use for indexing the rows.

Reply:
I have some changes here but I think the best approach for now is to leave this as is since handling arbitrary size could changes another functions too

Reply:
cc @felipeblazing @williamBlazing I will add this to my TODOs list